### PR TITLE
Resolve Documentation Collision for Extensions and Structs Sharing Names

### PR DESCRIPTION
**Pull request by rofle100lvl**
Resolves https://github.com/tuist/tuist/issues/6246

### Short description 📝

The purpose of the pr is to make identifier of documented type accurate. Now, in some cases extensions and classes can have the same names and because of that we can see extension documentation instead of struct

### How to test the changes locally 🧐

mise run docs:generate-manifests-docs
mise run docs:dev

### Contributor checklist ✅

- [✅] The code has been linted using run `mise run lint:fix`
- [x(Not sure about tests on web)] The change is tested via unit testing or acceptance testing, or both
- [✅] The title of the PR is formulated in a way that is usable as a changelog entry
- [✅] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by fortmarek — APPROVED** [review]
Thanks! This makes sense to me, would wait for @pepicrft review before merging 🙂

**Review by pepicrft — APPROVED** [review]

**Comment by pepicrft** [comment]
@all-contributors add @rofle100lvl for docs

**Comment by pepicrft** [comment]
Thanks a lot for this one @rofle100lvl

**Pull request merged**
